### PR TITLE
Add Path.fromPoints method.

### PR DESCRIPTION
**Pull request by yanglinzhen**
Add Path.fromPoints method.

### CODE REVIEW

File: src/main/java/androidx/graphics/Path.kt [hunk 1]
```diff
@@ -116,3 +116,19 @@ inline infix fun Path.xor(p: Path): Path {
         op(p, Path.Op.XOR)
     }
 }
+
+/**
+ * Generate a closed shape from [points].
+ */
+@RequiresApi(19)
+inline fun Path.fromPoints(points: List<PointF>){
```
Comment (JakeWharton) [inline src/main/java/androidx/graphics/Path.kt]: With a name like `fromPoints` I would expect this to be a factory function but it's a mutator.

File: src/main/java/androidx/graphics/Path.kt [hunk 2]
```diff
@@ -116,3 +116,19 @@ inline infix fun Path.xor(p: Path): Path {
         op(p, Path.Op.XOR)
     }
 }
+
+/**
+ * Generate a closed shape from [points].
+ */
+@RequiresApi(19)
```
Comment (JakeWharton) [inline src/main/java/androidx/graphics/Path.kt]: What makes this 19+?

File: src/main/java/androidx/graphics/Path.kt [hunk 3]
```diff
@@ -116,3 +116,19 @@ inline infix fun Path.xor(p: Path): Path {
...
+inline fun Path.fromPoints(points: List<PointF>) {
```
Comment (JakeWharton) [inline src/main/java/androidx/graphics/Path.kt]: With a name like `fromPoints` I would expect this to be a factory function but it's a mutator.

File: src/main/java/androidx/graphics/Path.kt (hunk 2 above)
Comment (yanglinzhen) [reply src/main/java/androidx/graphics/Path.kt]: Copied from method above, I'll remove it.

File: src/main/java/androidx/graphics/Path.kt [hunk 4]
```diff
@@ -116,3 +116,18 @@ inline infix fun Path.xor(p: Path): Path {
         op(p, Path.Op.XOR)
     }
 }
+
+/**
+ * Create a path with closed shape from [points].
+ */
+inline fun createPathFromPoints(points: List<PointF>): Path {
+    return Path().apply {
+        if (points.size < 3) throw IllegalArgumentException("points' size should be over 2")
```
Comment (romainguy) [inline src/main/java/androidx/graphics/Path.kt]: Why should it have more than 2?

File: src/main/java/androidx/graphics/Path.kt [hunk 5]
```diff
@@ -116,3 +116,18 @@ inline infix fun Path.xor(p: Path): Path {
         op(p, Path.Op.XOR)
     }
 }
+
+/**
+ * Create a path with closed shape from [points].
+ */
+inline fun createPathFromPoints(points: List<PointF>): Path {
```
Comment (romainguy) [inline src/main/java/androidx/graphics/Path.kt]: This method only creates lines, this is of rather limited use especially for a Path. Canvas has a drawLines() method for instance that can take an array of x/y coordinates to draw the path.

File: src/androidTest/java/androidx/graphics/PathTest.kt [hunk 6]
```diff
@@ -152,4 +152,23 @@ class PathTest {
...
     }
+
+    @Test fun testPathFromPoints() {
+        val rect = RectF(0f, 0f, 40f, 40f)
+
+        val points = ArrayList<PointF>()
+        points.add(PointF(rect.left, rect.top))
+        points.add(PointF(rect.right, rect.top))
+        try {
+            val illegalPath = createPathFromPoints(points)
```
Comment (romainguy) [inline src/androidTest/java/androidx/graphics/PathTest.kt]: I don't understand what makes such a path "illegal"

File: src/androidTest/java/androidx/graphics/PathTest.kt [hunk 7]
```diff
@@ -152,4 +152,23 @@ class PathTest {
...
+            assertEquals(true, illegalPath.isConvex)
+        } catch (e: IllegalArgumentException) {
+            e.printStackTrace()
+        }
+        points.add(PointF(rect.right, rect.bottom))
+        points.add(PointF(rect.left, rect.bottom))
+        val legalPath = createPathFromPoints(points)
+
+        assertEquals(true, legalPath.isConvex)
```
Comment (romainguy) [inline src/androidTest/java/androidx/graphics/PathTest.kt]: Why such emphasis on the path being convex?

File: src/main/java/androidx/graphics/Path.kt (hunk 5 above)
Comment (romainguy) [inline src/main/java/androidx/graphics/Path.kt]: It doesn't sound efficient to first build a list of point to then create a path, when you can just create the path in the first place.

File: src/androidTest/java/androidx/graphics/PathTest.kt (hunk 6 above)
Comment (yanglinzhen) [reply src/androidTest/java/androidx/graphics/PathTest.kt]: At first I thought it should draw a contour by canvas, not a line. This is not necessarily illegal

File: src/main/java/androidx/graphics/Path.kt (hunk 5 above)
Comment (yanglinzhen) [reply src/main/java/androidx/graphics/Path.kt]: drawLines() does not draw a closed shape

File: src/androidTest/java/androidx/graphics/PathTest.kt (hunk 7 above)
Comment (yanglinzhen) [reply src/androidTest/java/androidx/graphics/PathTest.kt]: It was not quite strict, I have changed it.

File: src/main/java/androidx/graphics/Path.kt (hunk 5 above)
Comment (yanglinzhen) [reply src/main/java/androidx/graphics/Path.kt]: for the case I have a list of point already, and want to build a path.

Comment (romainguy) [reply src/main/java/androidx/graphics/Path.kt]: All you have to do is add an extra line at the end (it's all that Path does).

**Pull request closed without merging**